commit/plugin.go

```diff
@@ -346,6 +346,22 @@ func (p *Plugin) Outcome(
 }
 
 func (p *Plugin) Close() error {
+	var errs []error
+
+	if p.offchainCfg.RMNEnabled {
+		if p.rmnHomeReader != nil {
+			if err := p.rmnHomeReader.Close(); err != nil {
+				errs = append(errs, fmt.Errorf("failed to close RMNHome reader: %w", err))
+				p.lggr.Errorw("Failed to close RMNHome reader", "err", err)
+			}
+		} else {
+			p.lggr.Warn("RMNHome reader was nil during Close")
+		}
+	}
+
+	if len(errs) > 0 {
+		return fmt.Errorf("errors during Close: %v", errs)
+	}
 	return nil
```

Comment:
Use [errors.Join](https://pkg.go.dev/errors#Join) instead:

```suggestion
	return errors.Join(errs)
```